src/coreclr/jit/morph.cpp

```diff
@@ -14092,27 +14110,41 @@ void Compiler::fgMergeBlockReturn(BasicBlock* block)
             fgReturnCount--;
         }
 
+#ifdef SWIFT_SUPPORT
+        // If merging GT_SWIFT_ERROR_RET nodes, ensure the error operand is stored to the merged return error local,
+        // so the correct error value is retrieved in the merged return block.
+        if ((ret != nullptr) && ret->OperIs(GT_SWIFT_ERROR_RET))
+        {
+            assert(genReturnErrorLocal != BAD_VAR_NUM);
+            const DebugInfo& di = lastStmt->GetDebugInfo();
+            GenTree*         swiftErrorStore =
+                gtNewTempStore(genReturnErrorLocal, ret->gtGetOp1(), CHECK_SPILL_NONE, nullptr, di, block);
```

Comment:
```suggestion
            GenTree*         swiftErrorStore =
                gtNewTempStore(genReturnErrorLocal, ret->gtGetOp1());
```
(For `pAfterStmt == nullptr` these last parameters aren't relevant)